### PR TITLE
Added Community Runtime for building OLLAMA LLMs in CML

### PR DESCRIPTION
**Pull request by SuperEllipse**
OLLAMA (refer to www.ollama.com)  allows easy ways to set up Local LLMs such as Llama2. By default setting up ollama requires sudo access hence we cannot use default runtimes. This runtime addresses that need. 

This PR is submitted by vishrajagopalan@cloudera.com

### CODE REVIEW

File: ollama/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    pip3 config set install.user false && \
+    SETUPTOOLS_USE_DISTUTILS=stdlib pip3 install \
+        --no-cache-dir \
+        --no-warn-script-location \
+        -r requirements.txt && \
+    rm -rf /build
+
+
+ENV ML_RUNTIME_EDITOR="PBJ Workbench" \
+    ML_RUNTIME_EDITION="Standard" \
```
Comment (peterableda) [inline ollama/Dockerfile:131]: This is overridden below.

File: ollama/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    SETUPTOOLS_USE_DISTUTILS=stdlib pip3 install \
+        --no-cache-dir \
+        --no-warn-script-location \
+        -r /build/requirements.txt && \
+    rm -rf /build
+
+ENV ML_RUNTIME_JUPYTER_KERNEL_NAME="python3" \
+    ML_RUNTIME_DESCRIPTION="PBJ Workbench Python runtime provided by Cloudera"
+
+ENV ML_RUNTIME_EDITION="Nvidia GPU Ollama" \
```
Comment (peterableda) [inline ollama/Dockerfile:149]: NVIDIA is all capital, they are very particular about this. :)

File: ollama/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    ML_RUNTIME_DESCRIPTION="This Version Includes Cuda Libraries and Ollama" \
```
Comment (peterableda) [inline ollama/Dockerfile:150]: CUDA is all capital too.

File: ollama/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    ML_RUNTIME_CUDA_VERSION="11.8.0"
+
+
+
+
+ENV \
+    ML_RUNTIME_METADATA_VERSION=2 \ 
+    ML_RUNTIME_SHORT_VERSION="1.2" \
```
Comment (peterableda) [inline ollama/Dockerfile:158]: short version should be integer.

File: ollama/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Ollama PBJ CUDA Runtime in CML
+Ollama provides an easy mechanism to set up local selfhosted / private LLMs and multi-modal Models. This runtime setsup the Ollama binaries, the associated CUDA and recommended python libraries for accessing OLLAMA in CML.
```
Comment (peterableda) [inline ollama/README.md:2]: "setsup" typo

File: ollama/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Ollama PBJ CUDA Runtime in CML
+Ollama provides an easy mechanism to set up local selfhosted / private LLMs and multi-modal Models. This runtime setsup the Ollama binaries, the associated CUDA and recommended python libraries for accessing OLLAMA in CML. 
+
+## Why do we need this runtime?
+Ollama cannot be set up by a pip ihstall and binary installation on Linux requires sudo access which is not available by default in a CML Terminal for security reasons.
```
Comment (peterableda) [inline ollama/README.md:5]: "ihstall" typo

File: ollama/etc/Rprofile.site [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Site R configuration.
```
Comment (peterableda) [inline ollama/etc/Rprofile.site:1]: Why do we need the R files?

File: ollama/picture/AddingRuntimeinCML.jpg
Comment (peterableda) [inline ollama/picture/AddingRuntimeinCML.jpg:1]: Do you have a usage example instead?

**Review by peterableda — CHANGES_REQUESTED** [review]
Can you remove the files that you don't need?

File: ollama/Dockerfile (hunk 1 above)
Comment (SuperEllipse) [reply ollama/Dockerfile:131]: Removed ML_RUNTIME_EDITION

File: ollama/Dockerfile (hunk 4 above)
Comment (SuperEllipse) [reply ollama/Dockerfile:158]: fixed

File: ollama/picture/AddingRuntimeinCML.jpg
Comment (SuperEllipse) [reply ollama/picture/AddingRuntimeinCML.jpg:1]: The usage example is provided in the Read me. Copying below..

A build is available on dockerhub that can be accessed as follows ( needs docker login) by using the format docker.io/your-dockerimagename-name:tag

A prebuilt version has been uploaded to the Docker hub and can can be accessed as follows

docker.io/superellipse/cml-pbj-cuda-ollama:v1.2

File: ollama/etc/Rprofile.site (hunk 7 above)
Comment (SuperEllipse) [reply ollama/etc/Rprofile.site:1]: removed R files

File: ollama/README.md (hunk 6 above)
Comment (SuperEllipse) [reply ollama/README.md:5]: fixed

**Review by peterableda — APPROVED** [review]

**Pull request merged**